sigridci/sigridci/objective.py

```diff
@@ -0,0 +1,41 @@
+# Copyright Software Improvement Group
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from enum import Enum
+
+
+class ObjectiveStatus(Enum):
+    ACHIEVED = 1
+    IMPROVED = 2
+    STAGNANT = 3
+    UNKNOWN = 4
+
+
+class Objective:
+
+    @staticmethod
+    def determineStatus(feedback, options):
+        rating = feedback["newCodeRatings"].get("MAINTAINABILITY", None)
+        target = options.targetRating
+        baseline = feedback["baselineRatings"].get("MAINTAINABILITY", None)
+        before = feedback["changedCodeBeforeRatings"].get("MAINTAINABILITY", baseline)
```

Comment:
Why would you take the baseline as default? The only reason why `before` would be empty (so won't contain `MAINTAINABILITY`) is if there are only new files in the code checked. This is intentional so you can recognise this situation in which you should only accept ratings higher than or equal to the target (this is the standard for new files). So the default should be `None`. I also originally thought that was the intention of your `before != None` check, but I guess that was for the case where there is no previous snapshot and thus `baseline` is also None.